File: leetcode/27.移除元素.cpp
/*
 * @Author: Opynicus
 * @Date: 2022-04-07 19:38:40
 * @LastEditTime: 2022-04-07 19:45:05
 * @LastEditors: Opynicus
 * @Description: 
 * @FilePath: \undefinede:\JuniorWork\leetcode\27.移除元素.cpp
 * 可以输入预定的版权声明、个性签名、空行等
 */
/*
 * @lc app=leetcode.cn id=27 lang=cpp
 *
 * [27] 移除元素
 */

// @lc code=start
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int slow = 0, fast = 0;
        for (fast = 0; fast < nums.size(); fast++)
        {
            if (nums[fast] != val)
            {
                nums[slow] = nums[fast];
                slow++;
            }
            
        }
        return slow;
    }
};
// @lc code=end

File: leetcode/27.移除元素.go
/*
 * @Author: your name
 * @Date: 2022-01-08 11:37:06
 * @LastEditTime: 2022-01-08 11:42:55
 * @LastEditors: Please set LastEditors
 * @Description: 打开koroFileHeader查看配置 进行设置: https://github.com/OBKoro1/koro1FileHeader/wiki/%E9%85%8D%E7%BD%AE
 * @FilePath: \leetcode\27.移除元素.go
 */
/*
 * @lc app=leetcode.cn id=27 lang=golang
 *
 * [27] 移除元素
 */

// @lc code=start
func removeElement(nums []int, val int) int {
	cnt := 0
	nLen := len(nums)
	for i := 0; i < nLen; i++ {
		if nums[i] != val {
			nums[cnt] = nums[i]
			cnt++
		}
	}
	return cnt
}
// @lc code=end

